test/lyapunov.jl

```diff
@@ -175,51 +173,55 @@ end
     lorenztr = tr(jacobian(dq0TN)) #trace of Lorenz system Jacobian matrix
     @test lorenztr == -(1+σ+β)
 
-    xw, λw = lyap_taylorinteg(lorenz!, q0, t0:1.0:tmax, _order, _abstol; maxsteps=2)
+    trange = t0:1.0:tmax
+    xw, λw = lyap_taylorinteg(lorenz!, q0, trange, _order, _abstol; maxsteps=5)
     @test xw[1,:] == q0
-    @test size(xw) == (length(t0:1.0:tmax), 3)
-    @test size(λw) == (length(t0:1.0:tmax), 3)
+    @test size(xw) == (length(trange), 3)
+    @test size(λw) == (5+1, 3)
     @test prod(isnan.(xw[2:end,:]))
-    xw_, λw_ = lyap_taylorinteg(lorenz!, q0, t0:1.0:tmax, _order, _abstol, lorenz_jac!; maxsteps=2)
+    xw2, λw2 = lyap_taylorinteg(lorenz!, q0, collect(trange), _order, _abstol; maxsteps=5)
```

Comment:
I just performed some benchmarks, and it seems better (in terms of time and allocations) to use `vec(trange)` instead of `collect(trange)`. This may also be useful elsewhere in our code; the important thing to keep in mind is that `vec` reshapes to a vector, but shares the memory, so changing the "new" vector also changes the old array.

Reply:
This also applies somewhere below...

Reply:
Just to be sure how to proceed, should I try to replace the occurrences of `collect` in the code with `vec`?

Reply:
I am not quite sure, but at least it is worth testing. In the case marked (and other similar) there is no risk that `trange` be changed, so I would simply test what happens if this is changed. Other occurrences of `collect` should be carefully addressed.

Reply:
Actually almost all occurrences of collect were in this same file (test/lyapunov.jl), so I went ahead and changed it